scripts/redis/gpt5_cache_migration.py

```
@@ -0,0 +1,385 @@
+            # Transform key
+            new_key = self.transformer.transform_key(old_key)
+
+            # Skip if key already migrated
+            if new_key == old_key:
+                self.stats.keys_skipped += 1
+                return False
+
+            # Get original value and TTL
+            value = await self.redis_client.get(old_key)
+            if value is None:
+                logger.warning(f"Key not found during migration: {old_key}")
```

Ruff (0.14.4)

Logging statement uses f-string

(G004)